src/components/application_manager/src/resumption/resumption_data_processor.cc

```diff
@@ -0,0 +1,1125 @@
+      !saved_app[strings::application_commands].empty() ||
+      !saved_app[strings::application_choice_sets].empty() ||
+      !saved_app[strings::windows_info].empty();
+
+  LOG4CXX_DEBUG(logger_,
+                std::boolalpha << "Application has data to restore: "
+                               << has_data_to_restore);
+
+  return has_data_to_restore;
+}
+
+bool ResumptionDataProcessor::HasGlobalPropertiesToRestore(
+    const smart_objects::SmartObject& saved_app) const {
+  LOG4CXX_AUTO_TRACE(logger_);
+
+  const smart_objects::SmartObject& global_properties =
+      saved_app[strings::application_global_properties];
+
+  const bool has_gl_props_to_restore =
+      !global_properties[strings::help_prompt].empty() ||
+      !global_properties[strings::keyboard_properties].empty() ||
+      !global_properties[strings::menu_icon].empty() ||
+      !global_properties[strings::menu_title].empty() ||
+      !global_properties[strings::timeout_prompt].empty() ||
+      !global_properties[strings::vr_help].empty() ||
+      !global_properties[strings::vr_help_title].empty();
+
+  LOG4CXX_DEBUG(logger_,
+                std::boolalpha
+                    << "Application has global properties to restore: "
+                    << has_gl_props_to_restore);
+
+  return has_gl_props_to_restore;
+}
+
+bool ResumptionDataProcessor::HasSubscriptionsToRestore(
+    const smart_objects::SmartObject& saved_app) const {
+  LOG4CXX_AUTO_TRACE(logger_);
+
+  const smart_objects::SmartObject& subscriptions =
+      saved_app[strings::application_subscriptions];
+
+  const bool has_ivi_subscriptions =
+      !subscriptions[strings::application_vehicle_info].empty();
+
+  const bool has_button_subscriptions =
+      !(subscriptions[strings::application_buttons].length() == 1 &&
+        static_cast<hmi_apis::Common_ButtonName::eType>(
+            subscriptions[strings::application_buttons][0].asInt()) ==
+            hmi_apis::Common_ButtonName::CUSTOM_BUTTON);
+
+  const bool has_waypoints_subscriptions =
+      subscriptions[strings::subscribed_for_way_points].asBool();
+
+  const bool has_appservice_subscriptions =
+      subscriptions.keyExists(app_mngr::hmi_interface::app_service) &&
+      !subscriptions[app_mngr::hmi_interface::app_service].empty();
+
+  const bool has_system_capability_subscriptions =
+      subscriptions.keyExists(strings::system_capability) &&
+      !subscriptions[strings::system_capability].empty();
+
+  const bool has_subscriptions_to_restore =
+      has_ivi_subscriptions || has_button_subscriptions ||
+      has_waypoints_subscriptions || has_appservice_subscriptions ||
+      has_system_capability_subscriptions;
+
+  LOG4CXX_DEBUG(logger_,
+                std::boolalpha << "Application has subscriptions to restore: "
+                               << has_subscriptions_to_restore);
+
+  return has_subscriptions_to_restore;
+}
+
+bool ResumptionRequestIDs::operator<(const ResumptionRequestIDs& other) const {
```

Comment:
move it upper as util function

Reply:
Done in eb410750791e0a6b8d8d4090b1ce53f1d547c0fd